pkg/porter/porter.go

```diff
@@ -67,6 +73,79 @@ func NewWithConfig(c *config.Config) *Porter {
 	}
 }
 
+// Initialize must be called before running Porter.
+func (p *Porter) Initialize() error {
+	if err := p.LoadData(); err != nil {
+		return err
+	}
+
+	if !p.Config.SkipPermissionCheck {
+		return p.CheckFilePermissions()
+	}
+
+	return nil
+}
+
+// CheckFilePermissions for files in PORTER_HOME and
+// return an error if they are less restrictive than recommended.
+func (p *Porter) CheckFilePermissions() error {
+	home, _ := p.GetHomeDir()
+	if p.Debug {
+		fmt.Fprintf(p.Err, "Checking file permissions in %s\n", home)
+	}
+	// Warn if any files that may contain sensitive information have the wrong permissions
+	wantMode := os.FileMode(0600)
+
+	checkFile := func(f string) error {
+		info, err := p.FileSystem.Stat(f)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return nil
+			}
+			return errors.Wrapf(err, "could not check file permissions on %s", f)
+		}
+
+		perm := info.Mode().Perm()
+		if wantMode != perm|wantMode {
+			// %o prints the octal form of the permission bits, e.g. 755.
+			return errors.Errorf("incorrect file permissions on %s, it should be %o. Correct it manually or by running porter storage fix-permissions.", f, wantMode)
+		}
+
+		return nil
+	}
+
+	checkDir := func(dir string) error {
+		return p.FileSystem.Walk(dir, func(path string, info fs.FileInfo, err error) error {
+			if err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
+				return err
+			}
+
+			if info.IsDir() {
+				return nil
+			}
+
+			return checkFile(path)
+		})
+	}
+
+	if p.ConfigFilePath != "" {
+		if err := checkFile(p.ConfigFilePath); err != nil {
+			return err
+		}
+	}
+
+	dirs := []string{"claims", "outputs", "credentials", "parameters"}
```

Comment:
We are only going to hard stop and complain about permissions for files that could contain sensitive values. If the permissions on a mixin or the cache is a bit wider, that isn't a big deal.